src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +112,60 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. Check for application objects owned by the application
+        application_objects = self.get_application_objects()
+        if len(application_objects) > 0:
+            application_objects_str = self._application_objects_to_str(
+                application_objects
+            )
+            if cascade is True:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will be deleted:
+                            {application_objects_str}
+                        """
+                    )
+                )
+            elif cascade is False:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will not be deleted:
```

Comment:
ditto, but consider adding a line stating what happens to them. How are their ownership affected?

Reply:
We run the drop command even though we expect it to fail.
Hypothetically, if the drop is somehow successful, I would assume the ownership is transferred to the account or current role?

Reply:
You should verify and state it for the user, not make assumptions IMHO.

Reply:
WDYT?
```
$ snow app teardown --no-cascade
The following objects are owned by application spcs_na_5_gbloom:
(WAREHOUSE) SPCS_NA_5_GBLOOM_WH
Dropping application spcs_na_5_gbloom now.
╭─ Error ───────────────────────────────────────────────────────────────────────╮
│                                                                               │
│                                                                               │
│     Could not successfully execute the Snowflake SQL statements.              │
│                                                                               │
│                 drop application spcs_na_5_gbloom                             │
│                                                                               │
╰───────────────────────────────────────────────────────────────────────────────╯
```

`drop_generic_object()` should probably display the SQL error message instead of a generic failure text. We can consider doing it as part of different task.